providers/luadns/luadnsProvider.go

```diff
@@ -0,0 +1,213 @@
+	providers.CanUseCAA:              providers.Can(),
+	providers.CanUsePTR:              providers.Can(),
+	providers.CanUseSRV:              providers.Can(),
+	providers.CanUseSSHFP:            providers.Can(),
+	providers.CanUseTLSA:             providers.Can(),
+	providers.DocCreateDomains:       providers.Can(),
+	providers.DocDualHost:            providers.Can(),
+	providers.DocOfficiallySupported: providers.Can(),
+}
+
+func init() {
+	fns := providers.DspFuncs{
+		Initializer:   NewLuaDNS,
+		RecordAuditor: AuditRecords,
+	}
+	providers.RegisterDomainServiceProviderType("LUADNS", fns, features)
+}
+
+// NewLuaDNS creates the provider.
+func NewLuaDNS(m map[string]string, metadata json.RawMessage) (providers.DNSServiceProvider, error) {
+	l := &luadnsProvider{}
+	l.creds.email, l.creds.apikey = m["email"], m["apikey"]
+	if l.creds.email == "" || l.creds.apikey == "" {
+		return nil, fmt.Errorf("Missing LuaDNS email or apikey")
+	}
+
+	// Get a domain to validate authentication
+	if err := l.fetchDomainList(); err != nil {
+		return nil, err
+	}
+
+	return l, nil
+}
+
+// GetNameservers returns the nameservers for a domain.
+func (l *luadnsProvider) GetNameservers(domain string) ([]*models.Nameserver, error) {
+	if len(l.nameserversNames) == 0 {
+		l.fetchAvailableNameservers()
+	}
+	return models.ToNameserversStripTD(l.nameserversNames)
+}
+
+// ListZones returns a list of the DNS zones.
+func (l *luadnsProvider) ListZones() ([]string, error) {
+	if err := l.fetchDomainList(); err != nil {
+		return nil, err
+	}
+	zones := make([]string, 0, len(l.domainIndex))
+	for d := range l.domainIndex {
+		zones = append(zones, d)
+	}
+	return zones, nil
+}
+
+// GetZoneRecords gets the records of a zone and returns them in RecordConfig format.
+func (l *luadnsProvider) GetZoneRecords(domain string) (models.Records, error) {
+	domainID, err := l.getDomainID(domain)
+	if err != nil {
+		return nil, err
+	}
+	records, err := l.getRecords(domainID)
+	if err != nil {
+		return nil, err
+	}
+	existingRecords := make([]*models.RecordConfig, len(records))
+	for i := range records {
+		existingRecords[i] = nativeToRecord(domain, &records[i])
+	}
+	return existingRecords, nil
+}
+
+// GetDomainCorrections returns a list of corrections to update a domain.
+func (l *luadnsProvider) GetDomainCorrections(dc *models.DomainConfig) ([]*models.Correction, error) {
+	err := dc.Punycode()
+	if err != nil {
+		return nil, err
+	}
+	domainID, err := l.getDomainID(dc.Name)
+	if err != nil {
+		return nil, err
+	}
+	records, err := l.GetZoneRecords(dc.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	checkNS(dc)
+
+	// Normalize
+	models.PostProcessRecords(records)
+
+	var corrections []*models.Correction
+	var corrs []*models.Correction
+	if !diff2.EnableDiff2 {
+		differ := diff.New(dc)
+		_, create, del, mod, err := differ.IncrementalDiff(records)
+		if err != nil {
+			return nil, err
+		}
+
+		corrections := []*models.Correction{}
+		corrs := []*models.Correction{}
```

Comment:
staticcheck reports:

```
providers/luadns/luadnsProvider.go:128:3: this value of corrs is never used (SA4006)
```

I think staticcheck is wrong.  (Why? Because the code works!)  However, the error will go away if you remove line 128 and use `:=` on line 130, 135, 140.